File: objc/Mediator/Mediator/MyDialogBoxMediator.m
#import "MyDialogBoxMediator.h"

@implementation MyDialogBoxMediator

-(void)onChange:(Widget*)originator {
    if([originator isKindOfClass:[ListBox class]]) {
        ListBox* listBox = (ListBox*) originator;
        NSString* item = [listBox getSelectedItem];
        [_button setTitle:item];
        return;
    }
    
    if([originator isKindOfClass:[CheckBox class]]) {
        CheckBox* checkBox = (CheckBox*) originator;
        [_button setEnabledState:checkBox.state];
        return;
    }
}

-(void)setListBox:(ListBox*)listBox {
    _listBox = listBox;
}

-(void)setCheckBox:(CheckBox*)checkBox {
    _checkBox = checkBox;
}

-(void)setButton:(Button*)button {
    _button = button;
}

@end

File: swift/Mediator/Mediator/MyDialogBoxMediator.swift
import Foundation

public class MyDialogBoxMediator : DialogBoxMediator {
    var listBox: ListBox? = nil
    var checkBox: CheckBox? = nil
    var button: Button? = nil
    
    func onChange(_ originator: Widget) {
        if originator is ListBox {
            let listBox: ListBox = originator as! ListBox
            let item: String = listBox.getSelectedItem()
            button?.setTitle(item)
            return
        }
        
        if originator is CheckBox {
            let checkBox: CheckBox = originator as! CheckBox
            button?.setEnabledState(checkBox.state)
            return
        }
    }
}
